### DISCUSSION TITLE
How to set global custom_encoder?

### DISCUSSION THREAD

**Discussion by lalawila** [fastapi/fastapi#10201] [Questions] [answered]
### First Check

- [x] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from datetime import datetime
from fastapi import APIRouter
from fastapi.encoders import jsonable_encoder

# now
app = FastAPI()
# expected
# app = FastAPI(custom_encoder={
#            datetime: lambda dt: int(dt.timestamp())
#        })

router = APIRouter()

@router.get("hi")
def hi():
    result = {hi: datetime.now()}
    # now
    return jsonable_encoder(
        result,
        custom_encoder={
            datetime: lambda dt: int(dt.timestamp())
        }
    )
    # expected
    # return result

app.include_router(router)
```

### Description

Sorry for my english not good.

I want to convert the datetime to a timestamp.  

How to set global custom_encoder? The code above work, But I dont want to call jsonable_encoder in every request.

I have tried to set custom `default_response_class` when init app. By content have been encode.
```
class MyJSONEncoder(json.JSONEncoder):
    def default(self, obj):
        if isinstance(obj, datetime):
            return int(obj.timestamp())
        return super().default(obj)


class MyJSONResponse(JSONResponse):
    def render(self, content: typing.Any) -> bytes:
       print(content) # {"hi": "2023-09-03T21:38:09.630000+00:00"}  content have been encoder, I can do nothing to it
        return json.dumps(
            content,
            ensure_ascii=False,
            cls=MyJSONEncoder,
            allow_nan=False,
            indent=None,
            separators=(",", ":"),
        ).encode("utf-8")
app = FastAPI(default_response_class=MyJSONResponse)
```

### Operating System

Windows

### FastAPI Version

103

### Pydantic Version

2.0

### Python Version

3.11

**Comment by YuriiMotov** [comment, marked as answer]
No simple solution here.. And `custom_encoder` wouldn't solve the problem. For example, it will not work if you return Pydantic V2 model:

```python
from fastapi import FastAPI
from datetime import datetime
from fastapi import APIRouter
from fastapi.encoders import jsonable_encoder
from pydantic import BaseModel


class MyModel(BaseModel):
    a: datetime

app = FastAPI()

router = APIRouter()

@router.get("/hi")
def hi():
    result = MyModel(a=datetime.now())
    return jsonable_encoder(  # This will not convert MyModel.a to timestamp
        result,
        custom_encoder={
            datetime: lambda dt: int(dt.timestamp())
        }
    )

app.include_router(router)
```

So, the solution is to define the response models and specify there how fields should be serialized
